simulator/ns-3.35/src/point-to-point/model/switch-mmu.cc

```diff
@@ -233,8 +253,9 @@ SwitchMmu::SetEgressLosslessAlg(uint32_t alg) {
 uint64_t SwitchMmu::DynamicThreshold(uint32_t port, uint32_t qIndex, std::string inout, uint32_t type) {
 	if (inout == "ingress") {
 		double remaining = 0;
-		if (ingressPool > ingressPoolUsed) {
-			uint64_t remaining = ingressPool - ingressPoolUsed;
+		if (ingressPool > ingressTotalUsed) {
```

Comment:
This seems incorrect to me. Even ingressTotalUsed (shared pool + headroom pool) is smaller than ingressPool, some of queues may have started to use PFC headroom buffer

Reply:
I made a change later, forgot to ping! 

```c++

if (inout == "ingress") {
		double remaining = 0;
		uint64_t ingressPoolUsed = ingressTotalUsed - xoffTotalUsed; // Total bytes used from the ingress pool specifically.
		if (ingressPool > ingressPoolUsed) {
			uint64_t remaining = ingressPool - ingressPoolUsed;
			// DT's threshold = Alpha x remaining.
			return std::min(uint64_t(alphaIngress[port][qIndex] * (remaining)), UINT64_MAX - 1024 * 1024);
		}
		else { // ingressPool is full. There is no `remaining` buffer in ingressPool. DT's threshold returns zero in this case, but using if else just to avoid threshold computations even in the simple case.
			return 0;
		}
	}
```